### PR TITLE
Fix: UXW-683 Saving document resets undo history

### PR DESCRIPTION
**Pull request by bpander**
Closes [UXW-683](https://linear.app/metabase/issue/UXW-683)

### Description

Fixes issue where undo history is cleared on save by remounting the DocumentPage when the entityId route param changes.

### How to verify

See [UXW-683](https://linear.app/metabase/issue/UXW-683) for repro steps.

Verify [UXW-564: CMD-Z on an existing Document removes all content (+ other edge case)](https://linear.app/metabase/issue/UXW-564/cmd-z-on-an-existing-document-removes-all-content-other-edge-case) is still fixed.

### Demo

https://github.com/user-attachments/assets/05759959-5eb4-426e-9509-ed5ffd69a221

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: enterprise/frontend/src/metabase-enterprise/documents/components/DocumentPage.tsx [hunk 1]
```diff
@@ -166,25 +165,6 @@ export const DocumentPage = ({
     return hasUnsavedChanges();
   });
 
-  // Reset state when document changes
```
Comment (bpander) [inline enterprise/frontend/src/metabase-enterprise/documents/components/DocumentPage.tsx:169]: ℹ️ This "Reset state when document changes" stuff is all handled in routes.tsx now. It remounts DocumentPage when the id changes. It might be heavy-handed but maybe this is a scenario where we _want_ to be heavy-handed?

I don't think we're always gonna remember to update whatever we need to update in this useEffect. Case in point, the comments sidebar stays open when you navigate to a new document (but it's fixed by remounting).

https://github.com/user-attachments/assets/5d86ff0e-abcb-46db-a697-933f8734c693

**Review by alxnddr — APPROVED** [review]
